cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/k8s/SparkContainerDriverLauncher.java

```diff
@@ -0,0 +1,208 @@
+import io.cdap.cdap.api.plugin.Plugin;
+import io.cdap.cdap.app.runtime.Arguments;
+import io.cdap.cdap.app.runtime.ProgramOptions;
+import io.cdap.cdap.app.runtime.spark.distributed.SparkContainerLauncher;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.guice.SupplierProviderBridge;
+import io.cdap.cdap.common.internal.remote.InternalAuthenticator;
+import io.cdap.cdap.common.internal.remote.RemoteClientFactory;
+import io.cdap.cdap.common.lang.jar.BundleJarUtil;
+import io.cdap.cdap.internal.app.ApplicationSpecificationAdapter;
+import io.cdap.cdap.internal.app.runtime.codec.ArgumentsCodec;
+import io.cdap.cdap.internal.app.runtime.codec.ProgramOptionsCodec;
+import io.cdap.cdap.internal.app.spark.ArtifactFetcherService;
+import io.cdap.cdap.internal.app.worker.sidecar.ArtifactLocalizer;
+import io.cdap.cdap.logging.guice.RemoteLogAppenderModule;
+import io.cdap.cdap.master.environment.MasterEnvironments;
+import io.cdap.cdap.master.spi.environment.MasterEnvironment;
+import io.cdap.cdap.master.spi.environment.MasterEnvironmentContext;
+import io.cdap.cdap.proto.id.ProgramId;
+import io.cdap.cdap.security.auth.context.AuthenticationContextModules;
+import io.cdap.cdap.security.guice.CoreSecurityModule;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.filesystem.LocalLocationFactory;
+import org.apache.twill.filesystem.Location;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Spark container launcher for launching spark drivers, and also allowing spark executors to fetch artifacts from it.
+ */
+public class SparkContainerDriverLauncher {
+  private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder())
+    .registerTypeAdapter(Arguments.class, new ArgumentsCodec())
+    .registerTypeAdapter(ProgramOptions.class, new ProgramOptionsCodec())
+    .create();
+  private static final String PROGRAM_JAR_NAME = "program.jar";
+  private static final String PROGRAM_JAR_EXPANDED_NAME = "program.jar.expanded.zip";
+  private static final String CDAP_APP_SPEC_KEY = "cdap.spark.app.spec";
+  private static final String PROGRAM_ID_KEY = "cdap.spark.program.id";
+  private static final String DEFAULT_DELEGATE_CLASS = "org.apache.spark.deploy.SparkSubmit";
+  private static final String DELEGATE_CLASS_FLAG = "--delegate-class";
+
+  //TODO CDAP-18504: Make sure compressed files are decompressed and then added to working dir.
+  private static final String WORKING_DIRECTORY = "/opt/spark/work-dir/";
+  private static final String CONFIGMAP_FILES_BASE_PATH = "/etc/cdap/localizefiles/";
```

Comment:
this class is already committed on develop. Only review change is `CONFIGMAP_FILES_BASE_PATH ` and copying files from `CONFIGMAP_FILES_BASE_PATH ` location to `WORKING_DIRECTORY `